File: src/dataset/python/anagram_exercism-ThomasZumsteg/gemini/GeminiFlash_anagram_v4.py
"""Finds anagrams of a word in a list of words"""

def detect_anagrams(anagram, word_list):
    """Finds any words in [word_list] that are anagrams of [anagram]"""
    anagram_lower = anagram.lower()
    sorted_anagram = tuple(sorted(anagram_lower))
    return [word for word in word_list
            if word.lower() != anagram_lower and tuple(sorted(word.lower())) == sorted_anagram]

File: src/dataset/Java/anagram_exercism-ThomasZumsteg/gemini/GeminiFlash_Anagram_v4.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Anagram {
    private final String anagram;
    private final String sortedAnagram;

    public Anagram(String anagram) {
        this.anagram = anagram;
        this.sortedAnagram = sortString(anagram.toLowerCase());
    }

    private String sortString(String str) {
        char[] chars = str.toCharArray();
        Arrays.sort(chars);
        return new String(chars);
    }

    public List<String> match(List<String> words) {
        List<String> anagrams = new ArrayList<>();
        String anagramLower = this.anagram.toLowerCase();

        for (String word : words) {
            String wordLower = word.toLowerCase();

            if (wordLower.length() != anagramLower.length() || anagramLower.equals(wordLower)) {
                continue;
            }

            if (sortedAnagram.equals(sortString(wordLower))) {
                anagrams.add(word);
            }
        }
        return anagrams;
    }
}